src/test/java/com/exasol/adapter/dialects/scalarfunction/VirtualSchemaTestTable.java

```diff
@@ -0,0 +1,39 @@
+package com.exasol.adapter.dialects.scalarfunction;
+
+import java.sql.SQLException;
+
+public class VirtualSchemaTestTable<T> {
+    private final String fullyQualifiedName;
+    private final ValueSetter<T> valueSetter;
+
+    /**
+     * Create a new instance of {@link VirtualSchemaTestTable}.
+     *
+     * @param fullyQualifiedName name of the table in the virtual schema
+     * @param valueSetter        function that can change the value of the single row of the table
+     */
+    public VirtualSchemaTestTable(final String fullyQualifiedName, final ValueSetter<T> valueSetter) {
+        this.fullyQualifiedName = fullyQualifiedName;
+        this.valueSetter = valueSetter;
+    }
+
+    public String getFullyQualifiedName() {
+        return this.fullyQualifiedName;
+    }
+
+    public void setValueOfSingleRow(final T value) throws SQLException {
+        this.valueSetter.setValueOfSingleRow(value);
+    }
+
+    @FunctionalInterface
+    public interface ValueSetter<T> {
```

Comment:
I think this approach is OK, but we definitely need a better name for this in interface (maybe `VirtualSchemaTableInitializer` or similar), and use a template method to provide the common implementation and the hooks to be extended.